tests/test_hubspot_pagination_test.py

```diff
@@ -1,59 +1,102 @@
 
-    def expected_page_size(self):
+    def get_properties(self):
         return {
-            # "subscription_changes": 10 - 1000,
-            "subscription_changes": 1000,
-            # "email_events": 10 - 1000,
-            "email_events": 1000,
-            # "forms": ??, #infinity
-            # "workflows": ??,
-            # "owners": ??,
-            # "campaigns": ??,
-            # "campaigns": 500, # Can't make test data
-            # "contact_lists": 20 - 250,  #count,
-            # "deal_pipelines": ?? , # deprecated
-            # "contacts_by_company": 100,  #count # deprecated
-            "contact_lists": 250,
-            "contacts": 100, #count
-            "companies": 250,
-            "deals": 100,
-            "engagements": 250,
+            'start_date' : datetime.strftime(datetime.today()-timedelta(days=7), self.START_DATE_FORMAT)
         }
 
+    def setUp(self):
+        self.maxDiff = None  # see all output in failure
+        set_up_start = time.perf_counter()
+
+        self.my_timestamp = self.get_properties()['start_date']
+        test_client = TestClient(self.my_timestamp)
+        existing_records = dict()
+        streams = self.expected_streams() - {'email_events'} # we get this for free with subscription_changes
+        limits = self.expected_page_limits()
+         # 'contacts_by_company' stream needs to get companyIds first so putting the stream last in the list
+        stream_to_run_last = 'contacts_by_company'
+        if stream_to_run_last in streams:
+            streams.remove(stream_to_run_last)
+            streams = list(streams)
+            streams.append(stream_to_run_last)
+
+        for stream in streams:
+            # Get all records
+            if stream == 'contacts_by_company':
+                company_ids = [company['companyId'] for company in existing_records['companies']]
+                existing_records[stream] = test_client.read(stream, parent_ids=company_ids)
+            elif stream in {'companies', 'contact_lists', 'subscription_changes', 'engagements'}:
+                existing_records[stream] = test_client.read(stream, since=self.my_timestamp)
+            else:
+                existing_records[stream] = test_client.read(stream)
+                # existing_records['subscription_changes'] = test_client.get_subscription_changes()  # see BUG_TDL-14938
+            # check if we exceed the limit
+            under_target = limits[stream] + 1 - len(existing_records[stream])
+            print(f'under_target = {under_target} for {stream}')
+
+            if under_target >= 0 :
+                print(f"need to make {under_target} records for {stream} stream")
+                #subscription changes will pass in records and under_target to get to the large limit quicker
+                if stream == "subscription_changes":
+                    test_client.create(stream, subscriptions=existing_records[stream], times=under_target)
```

Comment:
I don't think we want to make this call if under_target is zero. I think `if under_target >= 0` should change to `under_target > 0`